File: 0783-minimum-distance-between-bst-nodes/0783-minimum-distance-between-bst-nodes.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int min = Integer.MAX_VALUE;
    private Integer prev = null;
    public int minDiffInBST(TreeNode root) {
         inOrder(root);
        return min;
    }
    private void inOrder(TreeNode root){
        if(root==null) return;
        
        inOrder(root.left);
        
        if(prev!=null){
            min = Math.min(min,root.val-prev);
        }
        
        prev  = root.val;
        
        inOrder(root.right);
    }
}

File: 0783-minimum-distance-between-bst-nodes/0783-minimum-distance-between-bst-nodes.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */

var minDiffInBST = function(root) {
    
    let min = Number.MAX_VALUE;
    let prev = null;
    inOrder(root);
    
    function inOrder(root){
    if(root===null) return ;
    
    inOrder(root.left);
    
    if(prev!==null){
        min = Math.min(min,root.val-prev);
    }
    
    prev = root.val;
    
    inOrder(root.right);
   }
    
    return min;
};

